2.5/test/run

```diff
@@ -144,30 +146,55 @@ test_scl_usage() {
     echo "ERROR[exec /bin/sh -ic "${run_cmd}"] Expected '${expected}', got '${out}'"
     return 1
   fi
+  return 0
+}
+
+function test_scl_usage_functionality() {
+  test_scl_usage "ruby --version" "ruby ${RUBY_VERSION}."
+  check_result "$?" || return 1
+  return 0
+}
+
+function test_base_scl_usage() {
+  test_scl_usage
+  check_result "$?" || return 1
```

Comment:
If `check_result` is needed for every test, shouldn't the executor (not duplicating this in every test) run it?

Also from what I've understand `check_result` doesn't do much difference WRT handling the code. `return "$?"` essentialy serves the same purpose, isn't that right?

Reply:
let me check it, I am not sure.